ncm-puppet/src/main/perl/puppet.pm

```diff
@@ -1,29 +1,48 @@
-#${PMcomponent}
+# #
+# Software subject to following license(s):
+#   Apache 2 License (http://www.opensource.org/licenses/apache2.0)
+#   Copyright (c) Responsible Organization
+#
+
+# #
+# Current developer(s):
+#   Andrea Sartirana <sartiran@llr.in2p3.fr>
+#
+
+# 
+# #
+# puppet, 17.7.0, 1, Fri Aug 11 2017
+#
+
+package NCM::Component::puppet;
 
-use parent qw(NCM::Component);
-our $EC  = LC::Exception::Context->new->will_store_all;
+use strict;
+use warnings;
 
-use EDG::WP4::CCM::Path 16.8.0 qw(unescape);
+use NCM::Component;
+use vars qw(@ISA $EC);
+use base qw(NCM::Component);
+$EC  = LC::Exception::Context->new->will_store_all;
+
+use EDG::WP4::CCM::Path qw(escape unescape);
 
 use CAF::Process;
 use CAF::FileWriter;
 
 use Config::Tiny;
 use YAML::XS;
 
+$YAML::XS::QuoteNumericStrings = 0;
```

Comment:
why is this needed?

Reply:
hiera 5 requires a "version: 5" in the hiera.yaml file and with the default value of $YAML::XS::QuoteNumericStrings this would be rendered as "version: '5'" which causes a parsing error while running puppet apply. I've not find a way to toggle quoting/not-quoting item per item..

Reply:
does the version come from https://github.com/quattor/configuration-modules-core/pull/1201/files#diff-f4e590f8087d8658009187202140cc6aL11? a string is a string. perl does duck typing, getTree will do the right thing, but it's not garanteed within perl (eg as soon as you print a varaible holding an interger, the variabel becomes a string).
but you can get it right, or use TextRender. 

you don't have to chnage the schema, but at least add a comment here why this is required. also a unittest checing this would be good (i briefly looked at the unittests, i couldn't find one)

Reply:
ok, i had a quick look at the rest of the code. what you do with `unescapekeys` is not in textrender, so you can't use that. but changing the schema will probably do the right thing. (anyway, just my 2cents 😄

Reply:
No no, version is in hieraconf field which is a generic dict (type puppet_hieraconf) as it varies too much from version to version for having a fixed schema (as it was in the old version of the component). The "version" parameter seems to be correctly passed as a integer.  In the json I see
        "hieraconf": {
          "defaults": {
            "data_hash": "yaml_data",
            "datadir": "/etc/puppetlabs/code/environments/production/data"
          },
          "hierarchy": [
            {
              "name": "quattor",
              "path": "quattor.yaml"
            }
          ],
          "version": 5  <--- here
        },

But YAML:XS quoted such integers. I've not found better solutions that disabling integer quotes completely.

Reply:
ok, so the profile has correct type, but perl mangles it. my bet would be that this happens in unescape keys (it probably assigns the integer to a varaible that used to hold a string). i opened https://github.com/quattor/CCM/issues/178 so maybe in future you can use textrender to do this correctly.

Reply:
OK. Thanks. I'll add a comment and a unittest